File: Dev/C++/Mascaret/src/Tools/Network/HttpServer/httpConnection.cpp
#include <sstream>
#include <boost/algorithm/string.hpp>

#include "httpServer/httpConnection.h"
#include "httpServer/httpRequest.h"
#include "httpServer/httpServer.h"

HttpConnection::HttpConnection(boost::asio::io_service& io_service) : TcpConnection(io_service), _server(NULL)
{
}
HttpConnection::HttpConnection(boost::asio::io_service& io_service, HttpServer* server) : TcpConnection(io_service), _server(server)
{
}
HttpConnection::~HttpConnection()
{
}

void HttpConnection::start()
{
	HttpRequest* req=HttpRequest::createRequestFromSocket(&_socket);
	if(_server != NULL)
	{
		_server->manageRequest(req);
	}
	delete req;
}

File: Dev/CS/Mascaret/Mascaret/Tools/NetWork/HttpConnection.cs
using System.Collections;
using System.Net.Sockets;
using System.Net;

namespace Mascaret
{
    public class HttpConnection : TcpConnection
    {

        private HttpServer server;
        /*
        public HttpConnection(HttpServer server) : base()
        {
            this.server = server;
        }
        */

        /* WINPHONE */
        public HttpConnection(Socket socket, HttpServer server)
            : base(socket)
        {
            this.server = server;
        }


        public override void start()
        {
            HttpRequest request = null;
            /* WINPHONE */
            request = HttpRequest.createRequestFromSocket(socket);
            if (server != null)
            {
                server.manageRequest(request);
            }
        }

    }
}
